packages/core/src/odata/common/request/odata-request.ts

```diff
@@ -179,8 +171,7 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
       return {
         ...destinationRelatedHeaders,
         ...csrfHeaders,
-        ...defaultHeaders,
-        ...this.config.customHeaders
+        ...this.basicHeaders()
```

Comment:
https://github.com/SAP/cloud-sdk/pull/471#discussion_r494925157

Reply:
I understand and I am not happy with the name either. The reason I introduced that is, that I want to add these headers to the batch request and would have to construct them there as well (which was the case until now).

In my opinion this is not an option for batch:
```
return {
  accept: 'application/json',
  ...this.getETagHeader()
  ...this.getContentTypeHeader()
  ...this.config.customHeaders
}
```
You could argue that the destinationRelatedHeaders + CSRF headers are access related and the others aren't, so there is some kind of difference. This is also the reason why only the non-access headers are used in the batch sub request - access is only cleared once on the highest level.
Having one function to bundle them all might be slightly more future proof, because should we add other non-access headers it would also be reflected in the batch request immediately.
That said, I could imagine an option like this:
```
return {
...defaultHeaders(),
...customHeaders()
}
```
Following the naming in the implementation below.

Reply:
so what will go in the customHeaders() and what in the defaultHeaders() headers?

Reply:
```
return {
  accept: 'application/json', //default
  ...this.getETagHeader() //default
  ...this.getContentTypeHeader() //default
  ...this.config.customHeaders //custom
}
```